IdentityCore/src/network/response_serializer/MSIDAADResponseSerializer.m

```diff
@@ -27,11 +27,19 @@
 
 @implementation MSIDAADResponseSerializer
 
-- (id)responseObjectForResponse:(NSHTTPURLResponse *)httpResponse data:(NSData *)data error:(NSError **)error
+- (id)responseObjectForResponse:(NSHTTPURLResponse *)httpResponse
```

Comment:
In general, I'm a little bit concerned that response serializer returns id. Most of the serializers return NSDictionary, but some return e.g. NSURL. Somebody might think it's always NSDictionary and expect NSDictionary, but then they'll get a NSURL. Maybe we can discuss how we can improve it, but we can take it as a follow up. Let's not block this PR on that.

Reply:
I think it is fine to keep 'id' here, but lets discuss it offline :)